src/string_tensor_pack.hpp

```diff
@@ -5,20 +5,24 @@
 #pragma once
 
 #include <openvino/op/op.hpp>
+#include <openvino/op/string_tensor_pack.hpp>
 
 // Having a decomposed representation for a tensor, converts it to a single string tensor with element::string element type.
-class StringTensorPack : public ov::op::Op {
+class StringTensorPack : public ov::op::v15::StringTensorPack {
 public:
-    OPENVINO_OP("StringTensorPack");
+    OPENVINO_OP("StringTensorPack", "extension", ov::op::v15::StringTensorPack);
 
     StringTensorPack () = default;
 
     StringTensorPack(ov::OutputVector inputs, const std::string& mode = "begins_ends")
-        : ov::op::Op(inputs), m_mode(mode) {
+        : ov::op::v15::StringTensorPack(inputs[0], inputs[1], inputs[2]), m_mode(mode) {
         constructor_validate_and_infer_types();
     }
 
-    void validate_and_infer_types() override;
+    void validate_and_infer_types() override {
+        OPENVINO_ASSERT(m_mode == "begins_ends", "StringTensorPack supports only 'begins_ends' mode, but get ", m_mode);
+        ov::op::v15::StringTensorPack::validate_and_infer_types();
+    }
 
     std::shared_ptr<ov::Node> clone_with_new_inputs(const ov::OutputVector& inputs) const override {
         auto result = std::make_shared<StringTensorPack>(inputs, m_mode);
```

Comment:
please do not change this class. Just mark this as deprecated. The same thing needs with `StringTensorUnpack`

Reply:
deprecation in OV extensions is not populated to end users.

Reply:
this is populated for developers. We do want developers use op from opset so it should be sort of deprecated, meaning that we don't recommend to use it.
I don't understant why we should change it it create risks to break something. What are we going to reach by these changes?

Reply:
Moreover, since we have duplicating functionality in the core. Logically, we should declare when these extension ops will be removed in what release (like in 2026, for example)? It will keep tokenizers clean from useless stuff.

Reply:
The reason i relied on opset evaluate for `StringTensorPack` is that it't more optimized to run on CPU and also we will have exactly the same evaluate for both old and new IRs, and it's compatible with extension version

I would've used opset evaluate even for `StringTensorUnpack`, bur our extension op supports also single input and that would've break compatibility with old IRs.

Users cannot link their code against our lib they only can do add_extension, no one except ourselves in this repo is able to use this ops from extension. We discussed with @apaniukov yesterday deprecation. If we add deprecation macros `OPENVINO_DEPRECATED` we will have warning every time we compile. This information obfuscates real problems when we compile.

Reply:
I don't understand what confusion will be. In our products, we should move to opset ops that is logical step. User should also do this since they use 2025.0 release. So no warnings will be who keep their code up-to-date.
So let us introduce official deprecation (with macro) and no need to kepp duplicated functionality and spend double-effort to maintain two versions of this code.